var/spack/repos/builtin/packages/mark/package.py

```diff
@@ -0,0 +1,53 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+
+
+class Mark(Package):
+    """Program MARK, developed and maintained by Gary White (Colorado State
+    University) is the most flexible, widely used application currently
+    available for parameter estimation using data from marked individuals.
+
+    You will need to download the package yourself, unzip, rename it
+    following the guide in http://www.phidot.org/software/mark/rmark/linux/
+    Step(1). Spack will search your current directory for the download file.
+    Alternatively, add this file to a mirror so that Spack can find it.
+    For instructions on how to set up a mirror, see
+    http://spack.readthedocs.io/en/latest/mirrors.html"""
+
+    homepage = "http://www.phidot.org/software/mark/index.html"
+
+    version('1.0', '64f2fb0837c38d6e41c7546b25627146', expand=False)
```

Comment:
@adamjstewart @lingnanyuan  I was reading the docs above, and wondering if we could do something like (pseudo-code):
```python
if distro.id() == 'darwin':
    version('1.0', ..., url=http://www.phidot.org/software/mark/downloads/files/mark.osx.zip)
else:
    version('1.0', ..., url=http://www.phidot.org/software/mark/downloads/files/mark.64.zip)
```
That won't allow installing cross-installing (which is a rare occurrence imo), but at least it won't require people to download things manually.

Reply:
@alalazo See @2009 for my take on OS-specific downloads. In my opinion, we need to add a `when` parameter to the version directive to support this kind of thing if we want to do it right. `jdk` is another example of a Linux-only package until we implement this.

Reply:
But if they do have publicly available downloads, we should definitely use them instead of making users download them manually.

Reply:
Because when I tried spack create this url, spack reported error:
curl: (22) The requested URL returned error: 403 Forbidden
So I downloaded it manually.

Reply:
@adamjstewart or @alalazo  - any decision on how you want this done ? Lingnan has moved on, but we can make whatever changes are needed to get this merged.

Spack can't fetch the packge due to user-agent identification from curl. AFAIK - there's not a recommended way to do this from spack yet, right ?

Related issues I've seen are: 
https://github.com/spack/spack/issues/69
https://github.com/spack/spack/pull/6967